include/z64actor.h

```diff
@@ -581,4 +578,43 @@ typedef struct {
     /* 0x24 */ char unk_24[0x4];
 } NpcInteractInfo; // size = 0x28
 
+// Converts a number of bits to a bitmask, helper for params macros
+// e.g. 3 becomes 0b111 (7)
+#define NBITS_TO_MASK(n) \
+    ((1 << (n)) - 1)
+
+// Extracts the `n`-bit value at position `s` in `p`, shifts then masks
+// No possibility of sign extension
+#define PARAMS_GET(p, s, n) \
+    (((p) >> (s)) & NBITS_TO_MASK(n))
+
```

Comment:
feel like we can prob afford the input vars to have full names, just so theres no ambiguity. incase someone forgets the order of the args they can hover over it and be reminded very easily 

at least, id like if `p` was the whole word "params". I think its fine if s and n are single letters as they commonly are in things like gfx macros

Reply:
I wouldn't mind if the arguments were named the same as in `_SHIFTL(v, s, w)` actually